src/main/java/com/spotify/futures/CompletableFuturesExtra.java

```diff
@@ -55,4 +61,130 @@ private CompletableFuturesExtra() {
     }
     return new CompletableToListenableFutureWrapper<V>(future);
   }
+
+  public static <T> CompletableFuture<T> immediatelyFailed(Throwable throwable) {
+    final CompletableFuture<T> future = new CompletableFuture<T>();
+    future.completeExceptionally(throwable);
+    return future;
+  }
+
+  public static <T, U> CompletableFuture<U> handleCompose(CompletionStage<T> future,
+          final BiFunction<? super T, Throwable, CompletionStage<? extends U>> fn) {
+
+    final CompletableFuture<U> result = new CompletableFuture<U>();
+
+    future.whenComplete(new BiConsumer<T, Throwable>() {
+      @Override
+      public void accept(T value, Throwable throwable) {
+        final CompletionStage<? extends U> newStage;
+        try {
+          newStage = fn.apply(value, throwable);
+        } catch (Throwable e) {
+          result.completeExceptionally(e);
+          return;
+        }
+        if (newStage == null) {
+          result.completeExceptionally(new NullPointerException("fn returned null"));
+        } else {
+          newStage.whenComplete(new BiConsumer<U, Throwable>() {
+            @Override
+            public void accept(U value, Throwable throwable) {
+              if (throwable != null) {
+                result.completeExceptionally(throwable);
+              } else {
+                result.complete(value);
+              }
+            }
+          });
+        }
+      }
+    });
+    return result;
+  }
+
+  /**
+   * Returns a new CompletableFuture that is completed when this
+   * CompletableFuture completes, with the result of the given
+   * function of the exception triggering this CompletableFuture's
+   * completion when it completes exceptionally; otherwise, if this
+   * CompletableFuture completes normally, then the returned
+   * CompletableFuture also completes normally with the same value.
+   * Note: More flexible versions of this functionality are
+   * available using methods {@code whenComplete} and {@code handle}.
+   *
+   * @param fn the function to use to compute the value of the
+   * returned CompletableFuture if this CompletableFuture completed
+   * exceptionally
+   * @return the new CompletableFuture
+   */
+  public static <T> CompletableFuture<T> exceptionallyCompose(
```

Comment:
Regarding the types, my point is not that CompletableFuture is going to go away. It's that someone may add a 'better' (in some sense) CompletionStage that we'd prefer to return rather than returning CompletableFuture. This addition could happen in the JRE or elsewhere.